### DISCUSSION TITLE
Custom server and getStaticProps

### DISCUSSION THREAD

**Discussion by seloner** [vercel/next.js#21381] [Help] [answered]
Should revalidate option in getStaticProps work with custom server ?

**Comment by jamesmosier** [comment, marked as answer]
Yes. Ultimately the custom server is still running `next()` and has the same capabilities as running without a custom server.

**Reply by seloner** [reply]
Are you sure ? I am having issues . I am talking about this https://nextjs.org/docs/advanced-features/custom-server.
![image](https://user-images.githubusercontent.com/29483115/105320244-9ec47d80-5bce-11eb-80ea-1cd235ad14ad.png)

**Reply by jamesmosier** [reply]
I'm positive 😄  I have a custom server (based on [this example](https://github.com/vercel/next.js/tree/canary/examples/custom-server)) and this as one of my page components:

```js
export default function Home({ date }) {
  return (
    <div>
      Date is: {date}
    </div>
  );
}

export async function getStaticProps(context) {
  const date = new Date().toISOString();
  return {
    revalidate: 1,
    props: { date },
  };
}
```

I load the page and it shows the current date/time, I refresh it still shows the same date/time, I refresh again and it shows a new date/time.
